csrc/scheduler/matmul_utils.cpp

```diff
@@ -296,6 +270,44 @@ std::string isMatmulFusionDefinitionSupported(
     }
   }
 
+  // Check that canonical dim order is (B)MNK
+  // TODO: Remove this check once we are confident that non-standard orders are
+  // properly handled
+  {
+    std::vector<ValGroup> dim_ordering = mma_utils::canonicalDimOrdering(
+        tensor_roles, id_roles, permissive_graph);
+    VectorOfUniqueEntries<MatmulDomain> role_order;
+    for (const ValGroup& g : dim_ordering) {
+      const auto it = id_roles.find(g);
+      NVF_ERROR(it != id_roles.end());
+      role_order.pushBack(it->second);
+    }
+    NVF_ERROR(
+        role_order.size() == 3 || role_order.size() == 4,
+        "Expected either {B,M,N,K} roles or {M,N,K} but role_order.size()=",
+        role_order.size());
+    if (role_order.back() != MatmulDomain::K) {
+      return "Canonical dim order must be BMNK";
+    }
+    if (role_order.at(role_order.size() - 2) != MatmulDomain::N) {
+      return "Canonical dim order must be BMNK";
+    }
+    if (role_order.at(role_order.size() - 3) != MatmulDomain::M) {
+      return "Canonical dim order must be BMNK";
+    }
+
+    // Also check that dims within each role are consecutive with one another
+    // for this pattern.
+    // TODO: Lift this requirement by modifying the definition or setting
+    // allocation domains to support this setting in MmaOp
```

Comment:
I believe this condition is sufficient to avoid the problematic cases like the included test `MultipleNonConsecutiveNDims`.